### ISSUE TITLE
请问为什么要设置proj为nn.Parameter(torch.ones((1, 17, 129)) / 129)

### ISSUE THREAD

**Issue by Echhoo** [wzx99/CLIPOCR#4]
并且将clip的visual的pos_emb重新修改为p_e[:8+1],p_e[15:23]重新赋值

**Comment by wzx99** [comment]
1.对齐特征序图尺寸，vit和clip的特征图尺寸不一样，输入图像尺寸是(32,128),vit的patch_size是(4,8),特征序列长度为1+8\*16=129；clip的patch_size是(16,16)，特征序列长度为1+2\*8=17
2.clip的默认输入尺寸为(224,224)，为了能够直接处理(32,128)的输入图像，需要调整位置编码，代码里是相当于把(224,224)图像的前面(32,128)区域的位置编码进行了保留。保留其他区域的位置编码应该也行，可以自己拿识别图像和文本内容输入到clip里看看还能不能对齐

**Comment by Echhoo** [comment]
> 1.对齐特征序图尺寸，vit和clip的特征图尺寸不一样，输入图像尺寸是(32,128),vit的patch_size是(4,8),特征序列长度为1+8*16=129；clip的patch_size是(16,16)，特征序列长度为1+2*8=17 2.clip的默认输入尺寸为(224,224)，为了能够直接处理(32,128)的输入图像，需要调整位置编码，代码里是相当于把(224,224)图像的前面(32,128)区域的位置编码进行了保留。保留其他区域的位置编码应该也行，可以自己拿识别图像和文本内容输入到clip里看看还能不能对齐

_____________________________________________-------------------------
请问您损失函数的对比中，和l1损失对比时L1 loss的代码是：
1、
image_similar = image_features @ image_features.t()
    text_similar = text_features @ text_features.t()

    loss = l1_weight * F.l1_loss(image_similar, text_similar)
    
还是 2、
loss =F.l1_loss( image_features , text_features)
输入的特征维度都是[b*len,dim]去做计算吗？

**Comment by wzx99** [comment]
我记得是2，直接去做的

**Comment by Echhoo** [comment]
> 我记得是2，直接去做的

作者尝试过image尺寸大小吗？因为clip是利用224*224训练的。还是就默认使用32*128和4*8的patch了呢 十分感谢

**Comment by wzx99** [comment]
没训过224\*224的，就直接默认按照识别任务常用的尺寸（32\*128，4\*8）训的

**Comment by Echhoo** [comment]
> 没训过224*224的，就直接默认按照识别任务常用的尺寸（32*128，4*8）训的

我想问下您试过只使用clip的image encoder去做蒸馏吗？效果怎么样？

**Comment by wzx99** [comment]
跑过，表1里面就是用CLIP的不同模块进行蒸馏的
<img width="332" alt="QQ拼音截图20240229205258" src="https://github.com/wzx99/CLIPOCR/assets/58062105/7b392b6b-f06a-4255-a680-3b86ddf782df">
